pkg/controller/sparkapplication/driver_pod.go

```diff
@@ -0,0 +1,187 @@
+	}
+	if image == "" {
+		return "", "", fmt.Errorf("no image specified in .spec.image or .spec.driver.image in SparkApplication %s/%s",
+			app.Namespace, app.Name)
+	}
+
+	driverPodName := getDriverPodName(app)
+	submissionID := uuid.New().String()
+	submissionCmdArgs, err := buildSubmissionCommandArgs(app, driverPodName, submissionID)
+	if err != nil {
+		return "", "", err
+	}
+
+	command := []string{"sh", "-c", fmt.Sprintf("$SPARK_HOME/bin/spark-submit %s", strings.Join(submissionCmdArgs, " "))}
+
+	labels := map[string]string{
+		config.SparkAppNameLabel:            app.Name,
+		config.LaunchedBySparkOperatorLabel: "true",
+		config.SubmissionIDLabel:            submissionID,
+	}
+
+	imagePullSecrets := make([]corev1.LocalObjectReference, len(app.Spec.ImagePullSecrets))
+	for i, secret := range app.Spec.ImagePullSecrets {
+		imagePullSecrets[i] = corev1.LocalObjectReference{Name: secret}
+	}
+	imagePullPolicy := corev1.PullIfNotPresent
+	if app.Spec.ImagePullPolicy != nil {
+		imagePullPolicy = corev1.PullPolicy(*app.Spec.ImagePullPolicy)
+	}
+
+	var driverCpuQuantity string
+	if app.Spec.Driver.CoreRequest != nil {
+		driverCpuQuantity = *app.Spec.Driver.CoreRequest
+	} else {
+		driverCpuQuantity = fmt.Sprintf("%d", *app.Spec.Driver.Cores)
+	}
+
+	var driverCpuQuantityLimit string
+	if app.Spec.Driver.CoreLimit != nil {
+		driverCpuQuantityLimit = *app.Spec.Driver.CoreLimit
+	} else {
+		driverCpuQuantityLimit = driverCpuQuantity
+	}
+
+	driverOverheadMemory, err :=
+		resourceusage.MemoryRequiredForSparkPod(app.Spec.Driver.SparkPodSpec, app.Spec.MemoryOverheadFactor, app.Spec.Type, 1)
+	if err != nil {
+		return "", "", err
+	}
+
+	var args []string
+	for _, argument := range app.Spec.Arguments {
+		args = append(args, argument)
+	}
+
+	var envVars []corev1.EnvVar
+	for key, value := range app.Spec.Driver.EnvVars {
+		envVars = append(envVars, corev1.EnvVar{Name: key, Value: value})
+	}
+
+	for key, value := range app.Spec.SparkConf {
+		if strings.HasPrefix(key, "spark.kubernetes.driverEnv.") {
+			env := strings.ReplaceAll(key, "spark.kubernetes.driverEnv.", "")
+			envVars = append(envVars, corev1.EnvVar{Name: env, Value: value})
+		}
+	}
+
+	envVars = append(envVars,
+		corev1.EnvVar{Name: "SPARK_K8S_DRIVER_POD_IP",
+			ValueFrom: &corev1.EnvVarSource{FieldRef: &corev1.ObjectFieldSelector{FieldPath: "status.podIP"}}})
+
+	envVars = append(envVars,
+		corev1.EnvVar{Name: "SPARK_DRIVER_BIND_ADDRESS",
+			ValueFrom: &corev1.EnvVarSource{FieldRef: &corev1.ObjectFieldSelector{FieldPath: "status.podIP"}}})
+	envVars = append(envVars,
+		corev1.EnvVar{Name: "SPARK_DEBUG",
+			Value: "1"})
+	envVars = append(envVars, corev1.EnvVar{Name: "OPERATOR_START_TIME",
+		Value: strconv.FormatInt(time.Now().UnixNano()/1000000, 10)})
+
+	envVars = append(envVars,
+		corev1.EnvVar{Name: "SPARK_ENVIRONMENT",
+			Value: os.Getenv("SPARK_ENVIRONMENT")})
+
+	envVars = append(envVars,
+		corev1.EnvVar{Name: "SPARK_HOME",
+			Value: "/opt/spark"})
+
```

Comment:
Those values should not be explicitly set. Line 98 will add everything is in the CRD.
So the final list of environment variables should contain:
- Spec.Driver.Env 
- Spark config options prefixed by spark.kubernetes.driverEnv.